Make some methods of RequestCallback return void instead of bool

lw opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58753 Fix race condition in TP agent
* #58675 Make Future always extract DataPtrs in RPC tests
* #56863 Ensure async_execution works with CUDAFuture
* #57355 Avoid re-doing CUDA stream sync in OwnerRRef
* #58428 Make TP agent use streams from Future when sending response
* #58674 Set and propagate devices in RRef completion future
* #58427 Set streams when invoking UDFs
* #58426 Create CUDA-aware futures in RequestCallback
* #58425 Provide pre-extracted DataPtrs when completing a Future with a Message
* #58424 Allow Future::then to return pre-extracted DataPtrs
* #58423 Always use intrusive_ptr for Message (2 out of 2)
* #58422 Always use intrusive_ptr for Message (1 out of 2)
* #58421 Prevent using anything other than intrusive_ptr for Future
* #58420 Migrate remaining shared_ptr<Future> to intrusive_ptr
* #57861 Make remaining autograd methods return futures
* #57860 Make remaining RRef methods return futures
* #57859 Unify fetching RRefs
* #57858 Deduplicate Python object serialization
* #57857 Simplify process(Script|Python)(Remote)?Call
* #57856 Unify assignment of OwnerRRef result
* #57855 Make processPythonExecution return a future
* #57854 Remove getScriptRemoteCallType
* #57853 Simplify OwnerRRef completion
* #57852 Unify async execution for JIT functions
* #57851 Unify invoking JIT functions
* #57850 Unify invoking JIT operands
* **#57849 Make some methods of RequestCallback return void instead of bool**
* #57848 Centralize setting messageId in RequestCallback
* #57847 Make RequestCallback collect Futures from methods, rather than providing them
* #57846 Add helpers to manipulate futures
* #57638 Reduce overhead when Future invokes callbacks inline
* #57637 Introduce thenAsync method on Future

Some methods are currently returning bool, but I'll soon want them to return a Future. I could have them return a tuple of bool and Future, but that's a bit heavy. Instead it turns out we can very easily make them return void, which will simplify things.

Differential Revision: [D28224476](https://our.internmc.facebook.com/intern/diff/D28224476/)

lw committed:
Make some methods of RequestCallback return void instead of bool

Some methods are currently returning bool, but I'll soon want them to return a Future. I could have them return a tuple of bool and Future, but that's a bit heavy. Instead it turns out we can very easily make them return void, which will simplify things.

Differential Revision: [D28224476](https://our.internmc.facebook.com/intern/diff/D28224476/)

[ghstack-poisoned]

lw committed:
Update on "Make some methods of RequestCallback return void instead of bool"


Some methods are currently returning bool, but I'll soon want them to return a Future. I could have them return a tuple of bool and Future, but that's a bit heavy. Instead it turns out we can very easily make them return void, which will simplify things.

Differential Revision: [D28224476](https://our.internmc.facebook.com/intern/diff/D28224476/)

[ghstack-poisoned]

lw committed:
Update on "Make some methods of RequestCallback return void instead of bool"


Some methods are currently returning bool, but I'll soon want them to return a Future. I could have them return a tuple of bool and Future, but that's a bit heavy. Instead it turns out we can very easily make them return void, which will simplify things.

Differential Revision: [D28224476](https://our.internmc.facebook.com/intern/diff/D28224476/)

[ghstack-poisoned]

mrshenli commented on torch/csrc/distributed/rpc/request_callback_no_python.cpp:
nit: these can fit in two lines?

mrshenli reviewed (approved):
LGTM!

mrshenli commented on torch/csrc/distributed/rpc/request_callback_no_python.cpp:
Please feel free to update this in followup PRs. I don't want to make you rebase and wait for another 6H, just for these little changes.

lw committed:
Update on "Make some methods of RequestCallback return void instead of bool"


Some methods are currently returning bool, but I'll soon want them to return a Future. I could have them return a tuple of bool and Future, but that's a bit heavy. Instead it turns out we can very easily make them return void, which will simplify things.

Differential Revision: [D28224476](https://our.internmc.facebook.com/intern/diff/D28224476/)

[ghstack-poisoned]

lw committed:
Update on "Make some methods of RequestCallback return void instead of bool"


Some methods are currently returning bool, but I'll soon want them to return a Future. I could have them return a tuple of bool and Future, but that's a bit heavy. Instead it turns out we can very easily make them return void, which will simplify things.

Differential Revision: [D28224476](https://our.internmc.facebook.com/intern/diff/D28224476/)

[ghstack-poisoned]

lw committed:
Update on "Make some methods of RequestCallback return void instead of bool"


Some methods are currently returning bool, but I'll soon want them to return a Future. I could have them return a tuple of bool and Future, but that's a bit heavy. Instead it turns out we can very easily make them return void, which will simplify things.

Differential Revision: [D28224476](https://our.internmc.facebook.com/intern/diff/D28224476/)

[ghstack-poisoned]

lw committed:
Update on "Make some methods of RequestCallback return void instead of bool"


Some methods are currently returning bool, but I'll soon want them to return a Future. I could have them return a tuple of bool and Future, but that's a bit heavy. Instead it turns out we can very easily make them return void, which will simplify things.

Differential Revision: [D28224476](https://our.internmc.facebook.com/intern/diff/D28224476/)

[ghstack-poisoned]
